### DISCUSSION TITLE
Slow exit

### DISCUSSION THREAD

**Discussion by Baughn** [LazyVim/LazyVim#4356] [Q&A] [answered]
Something in the default configuration is causing nvim to pause for a second and a half before exiting. It seems potentially related to LSP: The problem is larger for Rust source code, but mostly nonexistent for Markdown, directories and such.

If I quit with `:noautocmd quit`, then the exit is instant. If I start a profiler first (`:profile start profile.log, :profile func *, :profile file *`), then the exit is unfortunately _also_ instant. I'm running a release build, with a stock lazyvim configuration, and I do not have any lazy-mounted filesystems, or network filesystems.

The buffer disappears almost instantly on :q, vim sits on an empty buffer for that second-and-a-half.

**Comment by dpetka2001** [comment, marked as answer]
Disable `noice.nvim` and see if that solves your problem.

**Reply by Baughn** [reply]
That worked. Is noice really a good idea for the default config?

**Reply by Baughn** [reply]
Looks likely to be https://github.com/folke/noice.nvim/issues/921

**Reply by dpetka2001** [reply]
Yes that's the open issue that seems relevant to what you're observing. You will have to wait until it gets fixed upstream.

**Reply by dpetka2001** [reply]
In the meantime until it gets fixed you can pin `noice.nvim` to `commit = "d9328ef"` that is also mentioned in the linked issue, that apparently is the last commit before that issue started happening.
